File: LeetCode/67.Add_Binary/67.Add_Binary.java
class Solution {
    public String addBinary(String a, String b) {
        StringBuilder sb = new StringBuilder();
        int i = a.length() - 1, j = b.length() - 1;
        int plus = 0, sum = 0;
        while (i >= 0 || j >= 0) {
            sum = plus;
            if (i >= 0) {
                sum += a.charAt(i) - '0';
                i -= 1;
            }

            if (j >= 0) {
                sum += b.charAt(j) - '0';
                j -= 1;
            }
            plus = sum > 1 ? 1 : 0;
            sb.append(sum % 2);
        }

        if (plus > 0)
            sb.append(plus);

        return sb.reverse().toString();
    }
}

File: LeetCode/67.Add_Binary/67.Add_Binary.go
func addBinary(a string, b string) string {
    var builder strings.Builder
    i, j := len(a)-1, len(b)-1
    plus, sum := 0,0

    for i>= 0 || j >= 0{
        sum = plus

        if i >= 0 {
            sum += int(a[i] - '0')
            i -= 1
        }

        if j >= 0 {
            sum += int(b[j] - '0')
            j -= 1
        }

        if sum > 1 {
            plus = 1
        }else{
            plus = 0
        }

        builder.WriteString(fmt.Sprintf("%d", sum%2))
    }

    if plus > 0 {
        builder.WriteString(fmt.Sprintf("%d", plus))
    }

    return reverseString(builder.String())
}

func reverseString(s string) string {
    runes := []rune(s)
    for i, j := 0, len(s)-1; i < j; i, j = i+1, j-1 {
        runes[i], runes[j] = runes[j], runes[i]
    }
    return string(runes)
}